House-of-glass/src/Contexts/Cart/CartContext.jsx

```diff
@@ -0,0 +1,169 @@
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  useEffect(() => {
+    getCart();
+  }, []);
+
+  const addToCart = async (product) => {
+    if (cartState.cart.some((item) => item._id === product._id)) {
+      incrementCart(product);
+    } else {
+      try {
+        console.log(header);
+        const response = await axios.post(
+          "/api/user/cart",
+          { product },
+          header
+        );
+
+        if (response.status === 201) {
+          cartDispatch({ type: "ADD_TO_CART", payload: response?.data?.cart });
+        }
+        Toast({
+          type: "success",
+          message: `${product.name} will be seeing you soon!`,
+        });
+      } catch (error) {
+        Toast({
+          type: "error",
+          message: "Something broke. Where's my glasses.",
+        });
+        console.log(error);
+      }
+    }
+  };
+
+  const deleteFromCart = async (product) => {
+    try {
+      const response = await axios.delete(
+        `/api/user/cart/${product._id}`,
+        header
+      );
+      if (response.status === 200) {
+        cartDispatch({
+          type: "DELETE_FROM_CART",
+          payload: response?.data?.cart,
+        });
+      }
+      Toast({
+        type: "success",
+        message: `${product.name} will not be seeing you soon!`,
+      });
+    } catch (error) {
+      Toast({
+        type: "error",
+        message: "Someone's here. Where's my glasses.",
+      });
+      console.log(error);
+    }
+  };
+
+  const incrementCart = async (product) => {
+    try {
+      const response = await axios.post(
+        `/api/user/cart/${product._id}`,
+        { action: { type: "increment" } },
+        header
+      );
+      if (response.status === 200) {
+        cartDispatch({ type: "INCREMENT_CART", payload: response?.data?.cart });
+      }
+      Toast({
+        type: "success",
+        message: `The ${product.name}'s will be seeing you soon!`,
+      });
+    } catch (error) {
+      Toast({
+        type: "error",
+        message: "Uh-Oh!. Where's my glasses.",
+      });
+      console.log(error);
+    }
+  };
+
+  const decrementCart = async (product) => {
+    if (cartState.cart.some((item) => item.qty === 1)) {
+      deleteFromCart(product);
+    } else {
+      try {
+        const response = await axios.post(
+          `/api/user/cart/${product._id}`,
+          { action: { type: "decrement" } },
+          header
+        );
+        if (response.status === 200) {
+          cartDispatch({
+            type: "DECREMENT_CART",
+            payload: response?.data?.cart,
+          });
+        }
+        Toast({
+          type: "success",
+          message: `The ${product.name}'s will not be seeing you.`,
+        });
+      } catch (error) {
+        Toast({
+          type: "error",
+          message: "Ouch!. Where's my glasses.",
+        });
+        console.log(error);
+      }
+    }
+  };
+
```

Comment:
Instead of making 2 separate functions for increment and decrement operations,you can instead go for writing a single function that takes the product details and the operation you want to perform and attach  it to the respective icons on the screen

Reply:
Yes , that is an option. I just prefer to keep them seperate for readability's sake. 
Don't want to make it look complicated.